packages/plugins/FileService/src/Worker/ipfs.ts

```diff
@@ -0,0 +1,65 @@
+import { Attachment } from '@dimensiondev/common-protocols'
+import { encodeText } from '@dimensiondev/kit'
+import { create, IPFSHTTPClient } from 'ipfs-http-client'
+import { isEmpty } from 'lodash-unified'
+import { landing } from '../constants'
+import urlcat from 'urlcat'
+import type { ProviderAgent, LandingPageMetadata, AttachmentOptions } from '../types'
+import { makeFileKeySigned } from '../helpers'
+
+function creatClient(): IPFSHTTPClient {
+    return create({
+        host: 'ipfs.infura.io',
+        port: 5001,
+        protocol: 'https',
+    })
+}
+
+class IPFSAgent implements ProviderAgent {
+    static providerName = 'IPFS'
+    client: IPFSHTTPClient
+    constructor() {
+        this.client = creatClient()
```

Comment:
It's still called at the app startup. Can you delay until its first use? e.g. in `makeAttachment`